cmd/memtierd/main.go

```diff
@@ -0,0 +1,397 @@
+	log.Infof("Updating container %s/%s/%s...", pod.GetNamespace(), pod.GetName(), ctr.GetName())
+
+	//
+	// This is the container update request handler. You can make changes to
+	// the container update before it is applied. Take a look at the functions
+	// in pkg/api/update.go to see the available controls.
+	//
+	// In addition to altering the pending update itself, you are also allowed
+	// to update other existing containers.
+	//
+
+	updates := []*api.ContainerUpdate{}
+
+	return updates, nil
+}
+
+func (p *plugin) PostUpdateContainer(pod *api.PodSandbox, ctr *api.Container) error {
+	log.Infof("Updated container %s/%s/%s...", pod.GetNamespace(), pod.GetName(), ctr.GetName())
+	return nil
+}
+
+func (p *plugin) StopContainer(pod *api.PodSandbox, ctr *api.Container) ([]*api.ContainerUpdate, error) {
+	log.Infof("Stopped container %s/%s/%s...", pod.GetNamespace(), pod.GetName(), ctr.GetName())
+
+	//
+	// This is the container (post-)stop request handler. You can update any
+	// of the remaining running containers. Take a look at the functions in
+	// pkg/api/update.go to see the available controls.
+	//
+
+	podName := pod.GetName()
+	dirPath := fmt.Sprintf("/tmp/memtierd/%s", podName)
+
+	// Kill the memtierd process
+	out, err := exec.Command("sudo", "pkill", "-f", dirPath).CombinedOutput()
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		if !ok || exitErr.ExitCode() != 1 {
+			// Error occurred that is not related to "no processes found"
+			log.Fatalf("Error killing memtierd process: %v. Output: %s\n", err, out)
+		} else {
+			// "No processes found" error, do nothing
+			log.Printf("No processes found for memtierd process\n")
+		}
+	}
+
+	err = os.RemoveAll(dirPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return []*api.ContainerUpdate{}, nil
+}
+
+func (p *plugin) RemoveContainer(pod *api.PodSandbox, ctr *api.Container) error {
+	log.Infof("Removed container %s/%s/%s...", pod.GetNamespace(), pod.GetName(), ctr.GetName())
+	return nil
+}
+
+func (p *plugin) onClose() {
+	log.Infof("Connection to the runtime lost, exiting...")
+	os.Exit(0)
+}
+
+func getFullCgroupPath(ctr *api.Container) []byte {
+	cgroupPath := ctr.Linux.CgroupsPath
+
+	split := strings.Split(cgroupPath, ":")
+
+	partOne := split[0]
+	partTwo := fmt.Sprintf("%s-%s.scope", split[1], split[2])
+
+	partialPath := fmt.Sprintf("%s/%s", partOne, partTwo)
+
+	fullPath := fmt.Sprintf("*/kubepods*/%s", partialPath)
+
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		log.Fatalf("failed to open /proc/mounts: %v", err)
+	}
+	defer file.Close()
+
+	cgroupMountPoint := ""
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		fields := strings.Fields(line)
+		if fields[0] == "cgroup2" {
+			cgroupMountPoint = fields[1]
+			break
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("failed to read /proc/mounts: %v", err)
+	}
+	findCmd := exec.Command("find", cgroupMountPoint, "-type", "d", "-wholename", fullPath)
```

Comment:
I think we should only use golang provided functionality here and not fire external program for this. Could we for example use `filepath.WalkDir` here?